static/js/publisher/release/revisionsList.js

```diff
@@ -21,11 +35,42 @@ export default class RevisionsList extends Component {
               </span>
             </span>
           </td>
+
+          <td className="u-align--right">
+            { canBeReleased &&
+              <button className="p-icon-button p-tooltip p-tooltip--btm-center" onClick={this.releaseClick.bind(this, revision)}>
+                &uarr;
+                <span className="p-tooltip__message">{`Promote to ${this.props.currentTrack}/edge`}</span>
+              </button>
+            }
+            { hasPendingRelease &&
+              <button className="p-icon-button p-tooltip p-tooltip--btm-center" onClick={this.undoClick.bind(this, revision)}>
+                &#x2715;
+                <span className="p-tooltip__message">Revert promoting this revision</span>
+              </button>
+            }
+          </td>
         </tr>
       );
     });
   }
 
+  releaseClick(revision) {
+    this.props.promoteRevision(revision, `${this.props.currentTrack}/edge`);
+  }
+
+  undoClick(revision) {
+    this.props.undoRelease(revision, `${this.props.currentTrack}/edge`);
+  }
+
+  isAlreadyReleased(revision) {
+    return Object.keys(this.props.releasedChannels).some((channel) => {
```

Comment:
`.some` - nice

Reply:
Oh yeah, now I feel smart, next week I'll not remember what the hell this thing does.